enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,560 @@
+A reader can find the proposed phase breakdown in [OCPPLAN-9429](https://issues.redhat.com/browse/OCPPLAN-9429).
+
+This document intends to describe the initial phases of this project. The proposed initial course of action:
+
+1. Update "openshift/api" with adding "External" PlatformType
+2. Ensure that all Red Hat operators tolerate the "External" platform and treat the same as the "None" platform
+
+Next phase which is out of the scope for this EP:
+
+3. Modify operators for doing specific things when seeing the "External" platform.
+   It will be briefly described in the [Affected Components](#affected-components) below.
+   However, this should be addressed in separate EPs on a per-component basis.
+
+#### Future work
+
+There are several topics in this area that would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and their concrete behaviour, for example, add a "capability" for machine-api
+2. Precisely define the reaction of the operators listed below for the "External" platform type
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+4. Research the necessity for engagement and api extension for ["on-prem"-like in-cluster network infrastructure](https://github.com/openshift/enhancements/blob/ce4d303db807622687159eb9d3248285a003fabb/enhancements/network/baremetal-networking.md) for the "External" platform.
+   This will depend on demand from partner cloud providers and their cloud capabilities (the presence of a load-balancer-like concept, for example).
+
+### Background
+
+At the moment, the [Infrastructure](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L14)
+resource serves as a primary source of information about the underlying infrastructure provider and provider-specific parameters.
+Specifically the [PlatformSpec](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L47)
+and [PlatformStatus](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L65) parts
+of the infrastructure resource.
+
+Given that PlatformSpec and PlatformStatus are defined as "discriminated unions" and are required to have the platform type encoded within
+"openshift/api" beforehand, it creates a significant footprint and plenty of effort from Red Hat engineers to create the initial technical enablement of
+a new cloud provider and mostly impossible without Red Hat engineering engagement.
+
+Since a lot of infrastructure related components (such as [CCCMO](https://github.com/openshift/cluster-cloud-controller-manager-operator), [Machine API operator](https://github.com/openshift/machine-api-operator), [Machine Config Operator](https://github.com/openshift/machine-config-operator), and so on)
```

Comment:
I know `so on` covers it but I suggest adding WMCO to this list especially given folks don't tend to think about Windows 😉

Reply:
added